packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,513 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+
+// Minimum padding from horizontal edges of segmented control to edges of
+// encompassing widget.
+const EdgeInsets _kHorizontalItemPadding = const EdgeInsets.symmetric(horizontal: 16.0);
+
+// Minimum height of the segmented control.
+const double _kMinSegmentedControlHeight = 28.0;
+
+// Light partially-transparent blue color. Used to fill the background of
+// a child option the user is temporarily interacting with through a long
+// press or drag.
+const Color _kPressedBackground = const Color(0x33007aff);
+
+/// An iOS-style segmented control.
+///
+/// Displays the widgets provided in the [Map] of [children] in a
+/// horizontal list. Used to select between a number of mutually exclusive
+/// options. When one option in the segmented control is selected, the other
+/// options in the segmented control cease to be selected.
+///
+/// A segmented control can feature any [Widget] as one of the values in its
+/// [Map] of [children]. The type T is the type of the keys used
+/// to identify each widget and determine which widget is selected. As
+/// required by the [Map] class, keys must be of consistent types
+/// and must be comparable. The ordering of the keys will determine the order
+/// of the widgets in the segmented control.
+///
+/// When the state of the segmented control changes, the widget calls the
+/// [onValueChanged] callback. The key associated with the newly selected
```

Comment:
The key --> The map key
(just to make sure they don't confuse it with a widget `Key`)